### ISSUE TITLE
Yet another Range Error: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by cmurczek** [jan-molak/serenity-js#62]
Hi,
when I run my tests I get
```
RangeError: Maximum call stack size exceeded
    at baseClone (C:\df\src\frontend\node_modules\lodash\lodash.js:2641:23)
    at C:\df\src\frontend\node_modules\lodash\lodash.js:2702:34
    at arrayEach (C:\df\src\frontend\node_modules\lodash\lodash.js:537:11)
    at baseClone (C:\df\src\frontend\node_modules\lodash\lodash.js:2696:7)
    at C:\df\src\frontend\node_modules\lodash\lodash.js:2702:34
    at arrayEach (C:\df\src\frontend\node_modules\lodash\lodash.js:537:11)
    at baseClone (C:\df\src\frontend\node_modules\lodash\lodash.js:2696:7)
    at cloneSet (C:\df\src\frontend\node_modules\lodash\lodash.js:4591:28)
    at initCloneByTag (C:\df\src\frontend\node_modules\lodash\lodash.js:6254:18)
    at baseClone (C:\df\src\frontend\node_modules\lodash\lodash.js:2680:20)`
```

and no report is being generated. 
I'm using

- serenity-js 1.5
- protractor 5.1.2
- cucumber 1.3.3

A workaround is to disable protractor's promise manager by adding 
`SELENIUM_PROMISE_MANAGER: false,`
to my protractor config. Not sure if this has any side effects, though.

**Comment by jan-molak** [comment]
Could this be related to #55 ?

**Comment by cmurczek** [comment]
Wow, you ARE quick in replying 👍 

I doubt it's related. So far I only have one assertion using
`const equals = <T>(expected: T) => (actual: T) => expect(actual).to.eventually.equal(expected);`
(and my scenario already fails during an earlier step :( )

**Comment by jan-molak** [comment]
No worries :-)

Would you be able to create an example project on github that demonstrates the issue?

The call stack only mentions lodash, so it's hard for me to tell what's causing the problem or to reproduce the issue.

**Comment by cmurczek** [comment]
I've uploaded the feature I referred to to https://github.com/cmurczek/serenity-test
Let me know if this works for you.
Note that I implemented a stage crew member called SetDesigner for setting up a mock backend. As soon as I get this working properly I plan on creating a PR for it. That is, if you were interested in it. :)

Thx for the help!

**Comment by cmurczek** [comment]
Have you been able to reproduce? The reason why my scenario fails is due to en error originating from actor.ts, where it tries to invoke an indexOf method of a member that is undefined. The member in question is a _frame_ of some sort. Could this be a follow up issue of having the PromiseManager deactivated? Should I open a separate issue for that?

**Comment by nbarrett** [comment]
Hi @cmurczek - could this be the same as what I've raised in #64?

**Comment by cmurczek** [comment]
Hi @nbarrett, yes this looks an awful lot like it! The line failing is the same, also the circumstances are similar. In my case it's a setup of a mockBackend that needs to be completed then( theActor.attemptsTo(...

**Comment by jan-molak** [comment]
Hey, @cmurczek! there are quite a few things going on in that sample repo... (different protractor configurations, dependencies, etc.)  Would you mind [narrowing down its scope](https://stackoverflow.com/help/mcve) to only those parts that allow reproducing the problem?

Can you please also let me know what is the command you're running in order to trigger the issue?

Thanks!
Jan

**Comment by jan-molak** [comment]
#64 should be fixed in `@serenity-js/core` 1.2.2

@cmurczek would you mind checking if that solves the problem you've observed?

**Comment by cmurczek** [comment]
Hi @jan-molak, like @nbarrett in #64 I could successfully verify the fix using serenity-js 1.7! Thanks for the great support!

**Comment by jan-molak** [comment]
My pleasure! Glad it helped 👍
